physics/sfcsub.F

```diff
@@ -4267,6 +4367,36 @@ subroutine analy(lugb,iy,im,id,ih,fh,len,lsoil,slmskl,slmskw,     &
         endif
       endif
 
+!
+!  soil color
+!
+      irtsoc=0
+      if(fnsoca(1:8).ne.'        ') then
+        call fixrda(lugb,fnsoca,kpdsoc,slmskl,
+     &             iy,im,id,ih,fh,socanl,len,iret
+     &,            imsk, jmsk, slmskh, gaus,blno, blto
+     &,            outlat, outlon, me)
+        irtsoc=iret
+        if(iret.eq.1) then
+          write(6,*) 'FATAL ERROR: soil color analysis read error.'
+          call abort
```

Comment:
I notice a bunch of `abort` calls throughout. CCPP schemes are not supposed to be able to kill the model. They're supposed to set the errflg and errmsg variables and return so that the host can exit gracefully.